src/components/list-item/react/ListItem.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { ReactElement } from 'react';
+
+import classNames from 'classnames';
+
+import { COMPONENT_PREFIX } from 'LumX/core/react/constants';
+
+import { IGenericProps, getRootClassName } from 'LumX/core/react/utils';
+import { handleBasicClasses } from 'LumX/core/utils';
+
+import { Theme, Themes } from 'LumX/components';
+
+/**
+ *  Authorized size values.
+ */
+const enum Sizes {
+    tiny = 'tiny',
+    regular = 'regular',
+    big = 'big',
+    huge = 'huge',
+}
+type Size = Sizes;
+
+/////////////////////////////
+
+/**
+ * Defines the props of the component.
+ */
+interface IListItemProps extends IGenericProps {
+    /* Whether the list item is selected or not */
+    isSelected?: boolean;
+    /* Whether the list item can be clicked */
+    isClickable?: boolean;
+    /* Component size*/
+    size?: Size;
+    /* before element */
+    before?: ReactElement;
+    /* after element */
+    after?: ReactElement;
+}
+type ListItemProps = IListItemProps;
+
+/////////////////////////////
+
+/**
+ * Define the types of the default props.
+ */
+interface IDefaultPropsType extends Partial<ListItemProps> {}
+
+/////////////////////////////
+//                         //
+//    Public attributes    //
+//                         //
+/////////////////////////////
+
+/**
+ * The display name of the component.
+ *
+ * @type {string}
+ * @constant
+ * @readonly
+ */
+const COMPONENT_NAME: string = `${COMPONENT_PREFIX}ListItem`;
+
+/**
+ * The default class name and classes prefix for this component.
+ *
+ * @type {string}
+ * @constant
+ * @readonly
+ */
+const CLASSNAME: string = getRootClassName(COMPONENT_NAME);
+
+/**
+ * The default value of props.
+ *
+ * @type {IDefaultPropsType}
+ * @constant
+ * @readonly
+ */
+const DEFAULT_PROPS: IDefaultPropsType = {
```

Comment:
`before` and `after` are set to optional, they should have default values

Reply:
To be discussed on Slack